lib/zip/output_stream.rb

```diff
@@ -57,7 +57,7 @@ def open(file_name, encrypter = nil)
       end
 
       # Same as #open but writes to a filestream instead
-      def write_buffer(io = ::StringIO.new(''), encrypter = nil)
+      def write_buffer(io = ::StringIO.new('').binmode, encrypter = nil)
```

Comment:
What if we pass an `IO` in here that hasn't been set to `binmode`? Do we still see the behaviour you're trying to avoid? This change only sets a newly created `StringIO` to `binmode`. Is this a problem?

How about:
```suggestion
      def write_buffer(io = ::StringIO.new(''), encrypter = nil)
        io.binmode if io.respond_to?(:binmode)
```
Which would match behaviour in `::Zip::File.open_buffer`?

Reply:
I found another location where the binmode would be set if supported. I think this is a cleaner solution:
https://github.com/henkeinfo/rubyzip/blob/5ce4e13ddd33443f01fa33c8208423b76d99fce3/lib/zip/file.rb#L151
Yes, you're right with your comment. I rewrote the `write_buffer` to behave like the source above and also fixed the same thing in yet another method.

Reply:
Yes I spotted that when I was looking through `File` and wondered why it was inconsistent. Thanks for the catch.

This looks good to me.